ingestion/src/metadata/utils/service_spec/service_spec.py

```diff
@@ -0,0 +1,91 @@
+"""
+Manifests are used to store class information
+"""
+
+from typing import Optional, Type, cast
+
+from pydantic import model_validator
+
+from metadata.generated.schema.entity.services.serviceType import ServiceType
+from metadata.ingestion.api.steps import Source
+from metadata.ingestion.models.custom_pydantic import BaseModel
+from metadata.utils.importer import get_module_dir, import_from_module
+
+
+class BaseSpec(BaseModel):
+    """
+    # The OpenMetadata Ingestion Service Specification (Spec)
+
+    This is the API for defining a service in OpenMetadata it needs to be in the classpath of the connector in
+    the form:
+
+    metadata.ingestion.source.{service_type}.{service_name}.service_spec.ServiceSpec
+
+    Example for postres:
+
+    metadata.ingestion.source.database.postgres.service_spec.ServiceSpec
+
+    You can supply either strings with the full classpath or concrete classes that will be converted to strings.
+
+    The use of strings for the values gives us a few advantages:
+    1. manifests can be defined using json/yaml and deserialized into this class.
+    2. We can dynamically import the class when needed and avoid dependency issues.
+    3. We avoid circular imports.
+    4. We can hot-swap the class implementation without changing the manifest (example: for testing).
+    """
+
+    profiler_class: Optional[str]
+    metadata_source_class: str
+
+    @model_validator(mode="before")
+    @classmethod
+    def transform_fields(cls, values):
+        """This allows us to pass in the class directly instead of the string representation of the class. The
+        validator will convert the class to a string representation of the class."""
+        for field in list(cls.model_fields.keys()):
```

Comment:
maybe some more thoughts on this:
1. if the only use case was about defining manifests as yaml/json -> do we even want to consider this at this point? maybe we can keep it simpler for first round and if the use case comes, then jump into it
2. if we end up needing this, some tests would be great

Reply:
This is what's being used de-facto in all our current OpenMetadata. Its tested in the integration and e2e tests. The only reason to use strings is to give developers the option to specify classes that are not part of the OpenMetadata module tree. We already do this for [CustomDatabaseConnection](https://github.com/open-metadata/OpenMetadata/blob/f0f74dbe5903bdace840320d68be5aa9c589a260/openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/customDatabaseConnection.json#L24) but we allow this only for the `Metadata Source` class. This will allow us to allow importing classes for other workflows.

Reply:
In any case implemented some unit tests